<launch>

<!-- env name="ROSCONSOLE_CONFIG_FILE" value="/home/davidz/work/ros_hydro_ws/src/sr_slam/launch/vro_console.config" / -->

<node pkg="sr_slam" type="test_vro" name="test_vro" output="screen">

  <!-- Visual Features, to activate GPU-based features see CMakeLists.txt -->
  <param name="config/feature_detector_type"         value="SIFTGPU"/><!-- If SIFTGPU is enabled in CMakeLists.txt, use SURF here -->
  <param name="config/feature_extractor_type"        value="SIFTGPU"/><!-- If SIFTGPU is enabled in CMakeLists.txt, use SURF here -->
  <param name="config/min_keypoints"                 value="20"/><!-- Extract no less than this many ... -->
  <param name="config/min_matches"                   value="12"/> <!-- Minimal number of matched features before RANSAC 29? -->
  <param name="config/nn_distance_ratio"             value="0.95"/> <!-- 0.5/ 0.7 for SIFT, 0.95 for SIFTGPU Feature correspondence is valid if distance to nearest neighbour is smaller than this parameter times the distance to the 2nd neighbour. This needs to be 0.9-1.0 for SIFTGPU w/ FLANN, since SIFTGPU Features are normalized-->

  <param name="config/max_dist_for_inliers"          value="0.05"/> <!-- 0.05 "Mahalanobis distance for matches to be considered inliers by ransac"-->
  <param name="config/ransac_iterations"             value="10000" /> <!-- 200 -->
  <param name="config/b_ransac_fixed_iter"           value="true"/> <!-- weather to use fixed   -->

  <!-- vro strategy -->
  <param name="config/vro_strategy"                 value="vro_plane_em" /> <!-- " VRO strategy: VRO_ori, VRO_my, VRO_plane, VRO_plane_em" -->

  <!-- vro parameters -->
  <!-- param name="vro_src_node_path"                   value="/home/davidz/work/data/SwissRanger4000/vro_results/dat/dataset_11" /-->
  <!-- param name="vro_tar_node_path"                   value="/home/davidz/work/data/SwissRanger4000/vro_results/dat/dataset_10" /-->
  <!-- param name="vro_src_node_path"     value="/home/davidz/work/data/SwissRanger4000/vro_results/bdat/angle_test/dataset_30" /-->
  <!-- param name="vro_tar_node_path"     value="/home/davidz/work/data/SwissRanger4000/vro_results/bdat/angle_test/dataset_39" /-->

  <!-- param name="vro_src_node_path"     value="/home/davidz/work/data/SwissRanger4000/vro_results/bdat/angle_test/pitch_v2/dataset_20" /-->
  <!-- param name="vro_tar_node_path"     value="/home/davidz/work/data/SwissRanger4000/vro_results/bdat/angle_test/pitch_v2/dataset_35" /-->

  <param name="vro_tar_node_path"     value="/home/davidz/work/data/SwissRanger4000/vro_results/bdat/angle_test/tx_v2/dataset_35"/>
  <param name="vro_src_node_path"     value="/home/davidz/work/data/SwissRanger4000/vro_results/bdat/angle_test/tx_v2/dataset_60"/>

  <param name="vro_match_result_log"     value="/home/davidz/work/data/SwissRanger4000/vro_results/bdat/angle_test/tx_v2/m_result_p25.log"/>
  <param name="vro_statistic_result_log" value="/home/davidz/work/data/SwissRanger4000/vro_results/bdat/angle_test/tx_v2/s_result_p25.log"/>

  <param name="plane_number_threshold"        value="0.1"/> <!-- minimal percent of points that deemed as a plane -->

  <param name="gt_tx"     value="0.25"/>
  <param name="gt_pitch"  value="0."/>
  <param name="gt_yaw"    value="0."/>

  <param name="vro_pack_number"                     value="500"/>
  <param name="vro_src_id"                          value="14" />
  <param name="vro_tar_id"                          value="14" />
  <param name="vro_single_match"                    value="false" /> <!-- true: match one pair, false: match 500 pairs -->
  <param name="vro_single_times"                    value="5" /> 
  <param name="vro_display_matches"                 value="false" /> <!-- weather to display the matched points -->
  <!--param name="vro_match_result_log"     value="/home/davidz/work/data/SwissRanger4000/vro_results/bdat/angle_test/pitch_v2/m_result_p15.log"/-->
  <!--param name="vro_statistic_result_log" value="/home/davidz/work/data/SwissRanger4000/vro_results/bdat/angle_test/pitch_v2/s_result_p15.log"/-->

  <!-- sift parameters -->
  <param name="sift_num_features"                   value="0" /> <!-- number of returned sift features -->
  <param name="sift_octave_layers"                  value="5" /> <!-- number octave layers -->
  <param name="sift_contrast_threshold"             value="0.04" /> <!-- initial threshold for keypint detection -->
  <param name="siftgpu_contrast_threshold"          value="0.0008" /> <!-- 0.0018 initial DOG threshold for siftgpu -->
  <param name="siftgpu_edge_threshold"              value="10." /> <!-- edge threshold for siftgpu  -->

 </node>
</launch>
